### ISSUE TITLE
confirmed value equals to unconfirmed at some point

### ISSUE THREAD

**Issue by vlddm** [kyuupichan/electrumx#900]
its like `{confirmed: 0, unconfirmed: 1000}` and then `{confirmed: 1000, unconfirmed: 1000} ` Next is `{confirmed: 1000, unconfirmed: 0}`
Looks like in the moment of first confirmation. Im using 1.12.0 on Litecoin mainnet.

**Comment by wedevelopio** [comment]
Bounty USD$100

**Comment by kyuupichan** [comment]
Yes it's not currently atomic; requests are served concurrently to block processing, and mempool processing is also concurrent.  I'm not sure if there is an easy fix; simply pausing all requests while a block is processed is not acceptable

**Comment by dmitry537** [comment]
Any fixes? This is critical.
